python/niizarr/_nii2zarr.py

```diff
@@ -320,7 +319,7 @@ def nii2zarr(inp, out, *,
         'chunks': c,
         'dimension_separator': r'/',
         'order': 'F',
-        'dtype': jsonheader['datatype'],
+        'dtype': jsonheader['DataType'],
```

Comment:
So that's where we'd have a problem if we use the JNifti type specification.
I would use JNifti types in the JSON header, and convert it to the zarr-compatible format here. It looks like it can be done very easily like this:
```python
np.dtype([("r", "uint8"), ("g", "uint8"), ("b", "uint8")]).descr
# >>> [('r', '|u1'), ('g', '|u1'), ('b', '|u1')]
```
Although we'd want to preserve the original endianness as well, so we may want to include the trick from line 110 here.

Reply:
Oh I realized probably we need to replace some endianness character on this line instead of just use the descr

Reply:
Yes, exactly. Similar to what we were doing earlier in the past